library/powersync/AttachmentQueue.ts

```diff
@@ -0,0 +1,406 @@
+import { AbstractPowerSyncDatabase, Transaction } from '@journeyapps/powersync-sdk-react-native';
+import * as FileSystem from 'expo-file-system';
+import { v4 as uuid } from 'uuid';
+import { ATTACHMENT_TABLE, AttachmentRecord, AttachmentState } from './AppSchema';
+import { AbstractStorageAdapter, UploadOptions } from '../storage/AbstractStorageAdapter';
+
+export const ATTACHMENT_LOCAL_STORAGE_KEY = 'attachments';
+export const ATTACHMENT_STORAGE_PATH = `${FileSystem.documentDirectory}${ATTACHMENT_LOCAL_STORAGE_KEY}`;
+export const ATTACHMENT_QUEUE_INTERVAL = 30_000;
+export const ATTACHMENT_CACHE_LIMIT = 100;
+
+export interface AttachmentQueueOptions {
+  powersync: AbstractPowerSyncDatabase;
+  storage: AbstractStorageAdapter;
+  getAttachmentIds?: () => Promise<string[]>;
+  syncInterval?: number;
+  cacheLimit?: number;
+}
+
+export class AttachmentQueue {
+  uploading: boolean;
+  downloading: boolean;
+  options: AttachmentQueueOptions;
+  downloadQueue: Set<string>;
+
+  constructor(options: AttachmentQueueOptions) {
+    this.uploading = false;
+    this.downloading = false;
+    this.options = {
+      ...options,
+      syncInterval: options.syncInterval ?? ATTACHMENT_QUEUE_INTERVAL,
+      cacheLimit: options.cacheLimit ?? ATTACHMENT_CACHE_LIMIT
+    };
+    this.downloadQueue = new Set<string>();
+  }
+
+  protected get powersync() {
+    return this.options.powersync;
+  }
+
+  protected get storage() {
+    return this.options.storage;
+  }
+
+  get table() {
+    return ATTACHMENT_TABLE;
+  }
+
+  async init() {
+    // Ensure attachment directory exists
+    await this.storage.makeDir(ATTACHMENT_STORAGE_PATH);
+
+    const attachmentIds = (await this.options.getAttachmentIds?.()) || [];
+    if (attachmentIds.length > 0) {
+      await this.markForSync(attachmentIds);
+    }
+
+    this.watchUploads();
+    this.watchDownloads();
+
+    setInterval(() => this.trigger(), ATTACHMENT_QUEUE_INTERVAL);
+  }
+
+  trigger() {
+    this.uploadRecords();
+    this.downloadRecords();
+    this.expireCache();
```

Comment:
Add some comments on why the periodic trigger is used in addition to the `watchUploads` + `watchDownloads`. I assume "retries" is one use case?